### PR TITLE
Initial read_parquet MVP implementation

### LINKED ISSUE #12: MVP: Wrapping read_parquet
**Feature request**
Blocks #9. Implement a wrapper for read_parquet that provides a more tailored interface for loading a "base" (main) parquet file and a set of additional parquet files that are automatically packed into nested columns in the base parquet file. E.g.
```
read_parquet(data="base.parquet", to_pack={"dia": "nested.parquet"})
```
**Before submitting**
Please check the following:

- [x] I have described the purpose of the suggested change, specifying what I need the enhancement to accomplish, i.e. what problem it solves.
- [x] I have included any relevant links, screenshots, environment information, and data relevant to implementing the requested feature, as well as pseudocode for how I want to access the new functionality.
- [x] If I have ideas for how the new feature could be implemented, I have provided explanations and/or pseudocode and/or task lists for the steps.

### PR DESCRIPTION
**Pull request by dougbrn**
## Change Description

- [x] My PR includes a link to the issue that I am addressing
Resolves #12.

## Solution Description

This PR implements a read_parquet method that reads a base file and a set of nested files using a to_pack dictionary kwarg. As I was implementing this, I noticed some friction with needing to specify kwargs for the nested frame loading in the same call as the base frame. For now, I focused on just getting column subsets implemented by using a dictionary to assign to each nested frame. This will get messier as we move to things like Dask and needing to potentially handle partitioning kwargs as well.

I thought of an alternative to this which I posted in the pie in the sky doc, which may scale better to more complex loading tasks, but maybe not important for now:
```
# csv instead of parquet but same idea
df = read_csv("objects.csv", columns=some_subset)
df = df.pack_csv("dia_sources.csv", "dia", columns=dia_subset)
df = df.pack_csv("forced_sources.csv", "forced", columns=forced_subset)
```

## Code Quality
- [x] I have read the Contribution Guide
- [x] My code follows the code style of this project
- [x] My code builds (or compiles) cleanly without any errors or warnings
- [ ] My code contains relevant comments and necessary documentation

### Bug Fix Checklist
- [ ] My fix includes a new test that breaks as a result of the bug (if possible)
- [ ] My change includes a breaking change
  - [ ] My change includes backwards compatibility and deprecation warnings (if possible)

### New Feature Checklist
- [x] I have added or updated the docstrings associated with my feature using the [NumPy docstring format](https://numpydoc.readthedocs.io/en/latest/format.html)
- [ ] I have updated the tutorial to highlight my new feature (if appropriate)
- [ ] I have added unit/End-to-End (E2E) test cases to cover my new feature
- [ ] My change includes a breaking change
  - [ ] My change includes backwards compatibility and deprecation warnings (if possible)

### Documentation Change Checklist
- [ ] Any updated docstrings use the [NumPy docstring format](https://numpydoc.readthedocs.io/en/latest/format.html)

### Build/CI Change Checklist
- [ ] If required or optional dependencies have changed (including version numbers), I have updated the README to reflect this
- [ ] If this is a new CI setup, I have added the associated badge to the README

### Other Change Checklist
- [ ] Any new or updated docstrings use the [NumPy docstring format](https://numpydoc.readthedocs.io/en/latest/format.html).
- [ ] I have updated the tutorial to highlight my new feature (if appropriate)
- [ ] I have added unit/End-to-End (E2E) test cases to cover any changes
- [ ] My change includes a breaking change
  - [ ] My change includes backwards compatibility and deprecation warnings (if possible)

### CODE REVIEW

File: src/nested_pandas/nestedframe/io.py [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+        Parquet library to use. If 'auto', then the option
+        ``io.parquet.engine`` is used. The default ``io.parquet.engine``
+        behavior is to try 'pyarrow', falling back to 'fastparquet' if
+        'pyarrow' is unavailable.
+
+        When using the ``'pyarrow'`` engine and no storage options are provided
+        and a filesystem is implemented by both ``pyarrow.fs`` and ``fsspec``
+        (e.g. "s3://"), then the ``pyarrow.fs`` filesystem is attempted first.
+        Use the filesystem keyword with an instantiated fsspec filesystem
+        if you wish to use its implementation.
```
Comment (hombit) [inline src/nested_pandas/nestedframe/io.py:50]: Let's always use `pyarrow` - it is our dependency anyway. We still may have this argument for compatibility, just raise for anything but `auto` and `pyarrow`

File: src/nested_pandas/nestedframe/io.py [hunk 2]
```diff
@@ -0,0 +1,69 @@
+# typing.Self and "|" union syntax don't exist in Python 3.9
+from __future__ import annotations
+
+import pandas as pd
+
+from .core import NestedFrame
+
+
+def read_parquet(
```
Comment (hombit) [inline src/nested_pandas/nestedframe/io.py:9]: Pandas has `dtype_backend`, but I'm not sure if we'd like to exposure it... We definitely should experiment with `dtype_backend='pyarrow'`, but later. Let's maybe have it here for the completeness and add some TODO for it?

**Review by hombit — APPROVED** [review]
Great, thank you!

File: src/nested_pandas/nestedframe/io.py (hunk 1 above)
Comment (dougbrn) [reply src/nested_pandas/nestedframe/io.py]: Sounds good, I opted to just use it implicitly and removed the kwarg just for simplicity

File: src/nested_pandas/nestedframe/io.py (hunk 2 above)
Comment (dougbrn) [reply src/nested_pandas/nestedframe/io.py:15]: I've added the kwarg as is, with the numpy_nullable default as per pandas. I added a todo to investigate the backend more

**Pull request merged**